### PR TITLE
Repeat survey missions without leaving interface

### PR DESCRIPTION
**Pull request by sparr**
#### Summary
Interface "Repeat survey missions without leaving the interface"

#### Purpose of change

Using the same survey mission to check multiple tiles currently requires many unnecessary steps, re-opening the interface, selecting the mission, etc.

#### Describe the solution

The mission prompts now run in a loop until the player presses escape or does a step that doesn't allow continuing.

#### Testing

I've played with both survey missions and tried different combinations of selecting tiles and list options, selecting invalid things, pressing escape and enter at various points.

#### Additional context
I also removed a bit of redundant prompting.
I recommend viewing the diff with whitespace changes hidden.

### CODE REVIEW

File: src/faction_camp.cpp [hunk 1]
```diff
@@ -4464,72 +4464,77 @@ bool basecamp::survey_field_return( const mission_id &miss_id )
     }
 
     popup( _( "Select a tile up to %d tiles away." ), 1 );
```
Comment (sparr) [inline src/faction_camp.cpp:4466]: This prompt will only appear once.

File: src/faction_camp.cpp [hunk 2]
```diff
@@ -4464,72 +4464,77 @@ bool basecamp::survey_field_return( const mission_id &miss_id )
...
-            finish_return( *comp, true, abort_msg, skill_construction.str(), 0 );
-            return true;
-        } else {
+    tripoint_abs_omt where( get_player_character().global_omt_location() );
+
+    while( true ) {
+        where = ui::omap::choose_point( where );
+        if( where == overmap::invalid_tripoint ) {
             return false;
         }
```
Comment (sparr) [inline src/faction_camp.cpp:4474]: The `return false` path is the way out of the loop without changing any state and is triggered by pressing escape while selecting a tile.

File: src/faction_camp.cpp [hunk 3]
```diff
@@ -4541,91 +4546,96 @@ bool basecamp::survey_return( const mission_id &miss_id )
...
-    }
 
-    const oter_id &omt_ref = overmap_buffer.ter( where );
-    const std::optional<mapgen_arguments> *maybe_args = overmap_buffer.mapgen_args( where );
-    const auto &pos_expansions = recipe_group::get_recipes_by_id( "all_faction_base_expansions",
-                                 omt_ref, maybe_args );
-    if( pos_expansions.empty() ) {
-        popup( _( "You can't build any expansions in a %s." ), omt_ref.id().c_str() );
```
Comment (sparr) [inline src/faction_camp.cpp:4574]: This message is repeated in the next prompt.

File: src/faction_camp.cpp (hunk 2 above)
Comment (kevingranade) [reply src/faction_camp.cpp:4474]: Is that stated in the UI?

Comment (sparr) [reply src/faction_camp.cpp:4474]: #77397 includes a feature to put a message/instructions at the top of the map tile choosing interface. If that would make this feel better, I can break it out as its own PR and get it merged so this PR can use it.

**Pull request merged**